_search-plugins/search-relevance/template-query.md

```diff
@@ -0,0 +1,215 @@
+
+```json
+POST /_plugins/_ml/models/mBGzipQB2gmRjlv_dOoB/_predict
+{
+  "parameters": {
+    "inputText": "happy moments"
+  }
+}
+```
+{% include copy-curl.html %}
+
+The response contains an `embedding` field with vector embeddings generated from the `inputText`: 
+
+```json
+{
+    "inference_results": [
+    {
+        "output": [
+        {
+            "name": "response",
+            "dataAsMap": {
+            "embedding": [
+                0.6328125,
+                0.26953125,
+                0.41796875,
+                -0.00579833984375,
+                1.859375,
+                0.2734375,
+                0.130859375,
+                -0.001007080078125,
+                0.138671875,
+                ...],
+            "inputTextTokenCount": 2
+            }
+        }
+        ],
+        "status_code": 200
+    }
+    ]
+}
+```
+ 
+### Step 1: Create an ingest pipeline
+
+Create an ingest pipeline to generate vector embeddings from text fields during document indexing. The `input_map` maps document fields to model inputs. In this example, the `text` source field in a document is mapped to the `inputText` field---the expected input field for the model. The `output_map` maps model outputs to document fields. In this example, the `embedding` output field from the model is mapped to the `text_embedding` destination field in your document:
+
+```json
+PUT /_ingest/pipeline/knn_pipeline
+{
+  "description": "knn_pipeline",
+  "processors": [
+    {
+      "ml_inference": {
+        "model_id": "Sz-wFZQBUpPSu0bsJTBG",
+        "input_map": [
+          {
+            "inputText": "text"
+          }
+        ],
+        "output_map": [
+          {
+            "text_embedding": "embedding"
+          }
+        ]
+      }
+    }
+  ]
+}
+```
+{% include copy-curl.html %}
+
+### Step 2: Index a document
+
+Index the following document into a `template-knn-1` index:
+
+```json
+PUT /template-knn-1/_doc/1
+{
+  "text": "red shoes"
+}
+```
+{% include copy-curl.html %}
+
+To view the document, send a GET request:
+
+```json
+GET /template-knn-1/_doc/1
+```
+{% include copy-curl.html %}
+
+The response shows that the embeddings generated by the model are stored in the `text_embedding` field along with the original `text`:
+
+```json
+{
+    "_index": "template-knn-1",
+    "_id": "1",
+    "_version": 2,
+    "_seq_no": 1,
+    "_primary_term": 1,
+    "found": true,
+    "_source": {
+    "text_embedding": [
+        -0.69140625,
+        0.8125,
+        0.51953125,
+        -0.7421875,
+        0.6875,
+        0.4765625,
+        -0.34375,
+        ...],
+    "text": "red shoes"
+    }
+}
+```
+
+### Step 3: Search using a template query
+
+Use the following template query to search the index. The `ml_inference` processor takes the input text `sneakers`, generates a vector embedding, replaces `${text_embedding}` with the generated vector, and searches for documents closest to the vector:
```

Comment:
"The `ml_inference` processor takes the input text `sneakers`, generates a vector embedding," => "The `ml_inference` processor generates a vector embedding from the input text `sneakers`,"?